src/main/scala/mesosphere/marathon/core/volume/VolumesModule.scala

```diff
@@ -0,0 +1,94 @@
+package mesosphere.marathon.core.volume
+
+import com.wix.accord._
+import mesosphere.marathon.core.volume.providers._
+import mesosphere.marathon.state._
+
+/**
+  * VolumeProvider is an interface implemented by storage volume providers
+  */
+trait VolumeProvider[+T <: Volume] {
```

Comment:
discussed this via hangout: quite possibly going to introduce plugins that can implement the VolumeProvider interface. in that case, a sealed trait would be limiting.